src/arrayMethodJoin.js

```diff
@@ -5,7 +5,38 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (separator === undefined) {
+      let str2 = this[0] + '';
+
+      for (let i = 1; i < this.length; i++) {
+        str2 += `,${this[i]}`;
+      }
+
+      return str2;
+    }
```

Comment:
you have almost the same loop later in your code, you don't need to use it here
besides, here you don't prevent `null` and `undefined`

<img width="291" alt="image" src="https://user-images.githubusercontent.com/55802630/177111970-12b73d6e-ab56-4d0b-9be7-ca5d20059638.png">